x-pack/libbeat/management/generate.go

```diff
@@ -171,8 +175,8 @@ func injectStreamProcessors(expected *proto.UnitExpectedConfig, inputType string
 	// for reasons I can't understand, as it just ends up shuffling it around
 	// to individual metricsets anyway, at least on metricbeat
 	if expectedID := expected.GetId(); expectedID != "" {
-		inputId := generateAddFieldsProcessor(mapstr.M{"input_id": expectedID}, "@metadata")
-		processors = append(processors, inputId)
+		inputID := generateAddFieldsProcessor(mapstr.M{"input_id": expectedID}, "@metadata")
```

Comment:
The comment on this is possibly obsolete because the AST transformations were removed in V2. Do we need to update the comment to reference what V2 currently does? Remember that this code will never interact with the V1 agent.

```go
	// the AST injects input_id at the input level and not the stream level,
	// for reasons I can't understand, as it just ends up shuffling it around
	// to individual metricsets anyway, at least on metricbeat
```

Reply:
Yah, this was a reference to the V1 behavior I was trying to emulate, might want to make that a bit more clear.